feedhenry/fh-js-sdk
# RHMAP-22002 : Unable to get the data returned previously

# Jira link(s)
https://issues.jboss.org/browse/RHMAP-22002

# What
Unable to get the data returned previously

# Why
Shows that the line in ajax.js with an workarround for IE6 stop to work properly with the upper version. However, we do not support so longer win platforms and IE6 browser in order to keep it anyway. 

<img width="1046" alt="screenshot 2018-11-27 at 15 13 38" src="https://user-images.githubusercontent.com/7708031/49091095-0bde6180-f257-11e8-99ad-2c37ab5d857e.png">

# How
Remove the workaround for IE6.  

# Verification Steps

## Test 1:
1. Create a Hello World web app project
2. Edit the packge.json file to use the version `3.0.12-beta-3`
3. Deploy in on live env
4. Open the dev browser. See [here](https://support-4556vpsffkmy7x5wwgbdlxph-live.mbaas2.eu.feedhenry.com/) the POC. 
5. Remove all after ? in the URL
6. Check if the error "Failed to load resource: the server responded with a status of 400 (Bad Request)" will occurs
7. Click on in the URL called on it and check if a result as follow will appear. 


```
{
--
"date": "Tue Nov 27 10:35:58 UTC 2018 - 1543314958045",
"msg": "Cloud App Connection is not 'ACTIVE'. Its status is 'DISABLED'. Connection GUID: drpfwyylvdgsvg243ylyalgj",
"status": "error"
}

```

See [here](https://support.eu.feedhenry.com/box/srv/1.1/app/init?_callback=jsonp1&_jsonpdata={%22cuid%22:%22CAB8C83E9DA44D948406DD658E5052DD%22,%22cuidMap%22:null,%22destination%22:%22web%22,%22sdk_version%22:%22FH_JS_SDK/2.18.6%22,%22appid%22:%224556vpsffkmy7x5wwgbdlxph%22,%22appkey%22:%2218403e7d8d571e7bf3d8ff7831999e5dfa3f6192%22,%22projectid%22:%224556vpv6tghex67czhruugiu%22,%22connectiontag%22:%220.0.2%22,%22__fh%22:{%22cuid%22:%22CAB8C83E9DA44D948406DD658E5052DD%22,%22cuidMap%22:null,%22destination%22:%22web%22,%22sdk_version%22:%22FH_JS_SDK/2.18.6%22,%22appid%22:%224556vpsffkmy7x5wwgbdlxph%22,%22appkey%22:%2218403e7d8d571e7bf3d8ff7831999e5dfa3f6192%22,%22projectid%22:%224556vpv6tghex67czhruugiu%22,%22connectiontag%22:%220.0.2%22}}) the result of this POC.

<img width="849" alt="screenshot 2018-11-27 at 15 03 56" src="https://user-images.githubusercontent.com/7708031/49090484-bb1a3900-f255-11e8-8424-86a72c49469b.png">
<img width="1463" alt="screenshot 2018-11-27 at 15 04 10" src="https://user-images.githubusercontent.com/7708031/49090490-bd7c9300-f255-11e8-9d85-4db972edae6b.png">



Following an image to illustrate how it is without the fix and after the version 3.18.6 of this SDK. 

<img width="820" alt="screenshot 2018-11-27 at 15 12 48" src="https://user-images.githubusercontent.com/7708031/49091158-26183f80-f257-11e8-9f7a-3a8075117164.png">

*NOTE* That is not possible click on in the GET rest and check the callback.

## Test 2:

1. Login in the domain
2. Create a new project based on Cordova Hello World Template with Hybrid Cordova client app only. See here
3. Add Distribution credentials. See [here](https://github.com/fheng/help/blob/e62db3b829e14758d16398ed72743cba2345cf8c/developer_guides/clientsdk/7.buildfarm_ios_certificates.md)
4. Edit the client app to use the dev tag to test of fh-js-sdk (`3.0.12-beta-3`) 
5. Go to the Build Tab and build this client app to use this cert. See [here](https://support.eu.feedhenry.com/#projects/4556vpv6tghex67czhruugiu/apps/4556vpwvmwgnn6zvmsahf25a/build) the POC.
6. Install this client app in iOS phone 
7. Initial the client app and add a text and click on in the button. All worked fine and the /hello endpoint was called with success in the cloud app
8. Go to Connections tag and disabled the tag used for this build. Please, add a text to check if it will be returned.
9. Close the client app and open it again
10. Fill the field and click on in the button. Check that an error message should appear since the connection tag. See the result of the POC.

![img_c8e907d26770-1](https://user-images.githubusercontent.com/7708031/49093565-45659b80-f25c-11e8-9387-d52f85a228ba.jpeg)


## Checklist:
